src/m365/commands/setup.spec.ts

```diff
@@ -0,0 +1,334 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { Cli } from '../../cli/Cli';
+import { CommandInfo } from '../../cli/CommandInfo';
+import { Logger } from '../../cli/Logger';
+import Command from '../../Command';
+import { telemetry } from '../../telemetry';
+import { CheckStatus, formatting } from '../../utils/formatting';
+import { pid } from '../../utils/pid';
+import { session } from '../../utils/session';
+import { sinonUtil } from '../../utils/sinonUtil';
+import commands from './commands';
+import { SettingNames } from './setup';
+import { interactivePreset, powerShellPreset, scriptingPreset } from './setupPresets';
+const command: Command = require('./setup');
+
+describe(commands.SETUP, () => {
+  let log: any[];
+  let logger: Logger;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+    (command as any).answers = {};
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      (command as any).configureSettings,
+      Cli.getInstance().config.set,
+      pid.isPowerShell
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
```

Comment:
Hi, just popping in here to say that we should refactor this `sinonUtil.restore([...` to `sinon.restore()`. This is a new way to ensure that the entire sandbox is cleared and will from now on be used in every command.